tools/stats-definition-exporter/main.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023-Present Couchbase, Inc.
+//
+// Use of this software is governed by the Business Source License included
+// in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+// in that file, in accordance with the Business Source License, use of this
+// software will be governed by the Apache License, Version 2.0, included in
+// the file licenses/APL2.txt.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/couchbase/sync_gateway/base"
+)
+
+const DefaultFullFilePath = "./metrics_metadata.json"
+
+func main() {
+	outputConsoleOnlyFlag := flag.Bool("no-file", false, "Output stat metadata to console (stdout) only.")
+	outputFileFlag := flag.String("output", DefaultFullFilePath, "Full file path of outputted JSON file if flag 'no-file' is false.")
+	flag.Parse()
+
+	logger := log.New(os.Stderr, "", 0)
+
+	stats, err := getStats(logger)
+	if err != nil {
+		logger.Fatalf("could not get stats: %v", err)
+	}
+
+	if *outputConsoleOnlyFlag {
+		err = statsToConsole(logger, stats)
+	} else {
+		err = statsToFile(logger, stats, *outputFileFlag)
+	}
+	if err != nil {
+		logger.Fatalf("could not write stats: %v", err)
+	}
```

Comment:
A lot of this code (and the associated sets of two functions for console/file) seem unnessesary, as the console can be considered the same as a file if your functions take an `io.Writer`